### PR TITLE
[dynamic child objects][cleanup] Remove creation of locks for child objects

### PR DESCRIPTION
**Pull request by tnowacki**
- Child objects are no longer valid transaction arguments
- They do not need locks? I think

### CODE REVIEW

File: crates/sui-core/src/authority.rs [hunk 1]
```diff
@@ -1199,7 +1199,7 @@ impl AuthorityState {
             ObjectInfoRequestKind::LatestObjectInfo(request_layout) => {
                 match self.get_object(&request.object_id).await {
                     Ok(Some(object)) => {
-                        let lock = if !object.is_owned_or_quasi_shared() {
+                        let lock = if !object.is_address_owned() {
                             // Unowned objects have no locks.
```
Comment (amnn) [inline crates/sui-core/src/authority.rs:1203]: Comment probably needs updating to account for the child object case.

**Review by amnn — APPROVED** [review]
> They do not need locks? I think

I think so too...

**Comment by mystenmark** [comment]
> They do not need locks? I think

For now, they don't, because our object store has read-after-write consistency. But this may not always be so. If we ever want to use eventually consistent object storage, we will have to have locks for child objects (owned locks will always require read-after-write). But that's not really something we need to worry about in this PR, there are many things that will have to change if we ever make that change.

**Comment by tnowacki** [comment]
> > They do not need locks? I think
> 
> For now, they don't, because our object store has read-after-write consistency. But this may not always be so. If we ever want to use eventually consistent object storage, we will have to have locks for child objects (owned locks will always require read-after-write). But that's not really something we need to worry about in this PR, there are many things that will have to change if we ever make that change.

@mystenmark, are you good with this landing then? And can you make sure I got all the spots I needed? I'm not very familiar with this part of the codebase.

**Comment by mystenmark** [comment]
@tnowacki i think there are a few other places that could use attention:

https://github.com/MystenLabs/sui/blob/826315b2fffb09011a25611f09df46a89850f119/crates/sui-core/src/authority/authority_store.rs#L549-L553

https://github.com/MystenLabs/sui/blob/826315b2fffb09011a25611f09df46a89850f119/crates/sui-core/src/authority/authority_store.rs#L596-L598

https://github.com/MystenLabs/sui/blob/826315b2fffb09011a25611f09df46a89850f119/crates/sui-core/src/authority/authority_store.rs#L634-L636

Most of these are special case paths for genesis or gateway. Probably nothing bad will happen if we ignore them, but it couldn't hurt just to filter child objects out in these cases as well.

**Review by mystenmark — APPROVED** [review]
approved with a few comments

**Comment by tnowacki** [comment]
Thanks for the pointers, @mystenmark! Updated with checks in 2/3 of those spots. The remaining spot could not be checked, but child objects should already be filtered (and commented as much).

**Pull request merged**